class.jetpack.php

```diff
@@ -669,6 +670,14 @@ function point_edit_links_to_calypso( $default_url, $post_id ) {
 		return esc_url( sprintf( 'https://wordpress.com/%s/%s/%d', $path_prefix, $site_slug, $post_id ) );
 	}
 
+	function point_edit_comment_links_to_calypso( $url ) {
+		wp_parse_str( wp_parse_url( $url, PHP_URL_QUERY ), $query_args );
```

Comment:
Could you walk me through what's happening here? Does `wp_parse_url` not return an array of the url parts?

Reply:
Sure. `PHP_URL_QUERY` [says](http://php.net/manual/en/function.parse-url.php), "just return the `query` key value of that array as a string" (it's the only part we're interested in). `wp_parse_str` (which [uses](http://php.net/manual/en/function.parse-str.php) `parse_str`) parses the query params it finds in that string and assigns them to the `$query_args` array. The one we want that contains the comment id is the `c` param (which is awkwardly assigned as `amp;c`).

I _could_ just use basic regex (`\d+$`?) to strip whatever number off the end of the string, but that feels a little more fragile over time.

Reply:
Does the caller not have access to the comment id here?

Reply:
I'd at the least add a quick comment since `amp;c` is a bit unexpected

Reply:
Sure, good idea; added in 948498f5bd092049dea575f3c564fa81060959d6.